menu.py

```diff
@@ -14,42 +16,51 @@ def copy_nodes(window, data=False):
     buff = []
     for node in nodes:
         if isinstance(node, Node):
-            buff.append({'type': 'node',
+            buff.append(OrderedDict({'type': 'node',
                          'x': node.x - x,
                          'y': node.y - y,
                          'size': node.editor_size,
                          'color': node.color,
                          'code': node.code,
                          'connects': node.get_con_id(),
-                         'parent': node.id})
+                         'parent': node.id}))
         elif isinstance(node, Field):
-            buff.append({'type': 'field',
+            buff.append(OrderedDict({'type': 'field',
                          'x': node.x - x,
                          'y': node.y - y,
                          'size': (node.w, node.h),
                          'code': node.document.text,
                          'connects': node.get_con_id(),
-                         'parent': node.id})
+                         'parent': node.id}))
         elif isinstance(node, Sub):
-            buff.append({'type': 'sub',
+            buff.append(OrderedDict({'type': 'sub',
                          'x': node.x - x,
                          'y': node.y - y,
                          'size': node.editor_size,
                          'color': node.color,
                          'code': node.code,
                          'connects': node.get_con_id(),
-                         'parent': node.id})
+                         'parent': node.id}))
     if data:
-        return str(buff)
-    pyperclip.copy(str(buff))
+        #return str(buff)
+        return json.dumps(buff, indent=4)
+    #pyperclip.copy(str(buff))
+    pyperclip.copy(json.dumps(buff, indent=4))
     window.info('Copy ' + str(len(buff)) + ' nodes')
 
 
 def paste_nodes(window, data=None):
     x, y = (0, 0) if data else (window.pointer[0], window.pointer[1])
     buff = []
     try:
-        paste = eval(data) if data else eval(pyperclip.paste())
+        #paste = eval(data) if data else eval(pyperclip.paste())
+        data = data if data else pyperclip.paste()
+        try:
+            paste = json.loads(data)
+            #paste = json.loads(data, object_pairs_hook=OrderedDict)
```

Comment:
Why not use the object_pairs_hook variant, I think this is the canonical way to do it?

Reply:
btw, great work on implementing this!

Reply:
May I just don't understand this, but I don't see the point in getting an OrderedDict when loading the data, it's of no use. I see the point in saving, in order to have a well ordered text/json file for diffs e.g. - and that is implemented.